pkg/configure/oneagent/pmc/ruxit/from.go

```diff
@@ -70,26 +78,22 @@ func FromConf(reader io.Reader) (ProcConf, error) {
 				Key:     strings.Trim(splitLine[0], whiteSpace),
 			}
 
-			if len(splitLine) == 2 {
+			if len(splitLine) == expectedConfigParts {
 				prop.Value = strings.Trim(splitLine[1], whiteSpace)
 			}
 
 			result = append(result, prop)
 		}
 	}
 
-	if err := scanner.Err(); err != nil {
-		return ProcConf{}, errors.WithStack(err)
-	}
-
-	return ProcConf{
-		Properties: result,
-		Revision:   0,
-	}, nil
+	return ProcConf{Properties: result}, scanner.Err()
 }
 
-func confSectionHeader(confLine string) string {
-	if matches := sectionRegexp.FindStringSubmatch(confLine); len(matches) != 0 {
+// confSectionHeader extracts the section header from a line, if present.
+func confSectionHeader(line string) string {
+	matches := sectionRegexp.FindStringSubmatch(line)
+
+	if len(matches) == expectedSectionParts {
```

Comment:
this logic different, than it was before, because `len(matches) != 0 ` gives us everything > 1 and then we return first. but now it's a bit different. May be it's better but it requires extra unit-testing .